internal/backend/grpc/auth.go

```diff
@@ -197,6 +226,65 @@ func (s *authServer) AwaitPublicKeyConfirmation(ctx context.Context, request *au
 	return &emptypb.Empty{}, nil
 }
 
+// RevokePublicKey deletes public key by ID.
+// Only authenticated user keys can be removed.
+func (s *authServer) RevokePublicKey(ctx context.Context, request *authpb.RevokePublicKeyRequest) (*emptypb.Empty, error) {
+	creds, err := auth.CheckGRPC(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if !creds.HasValidSignature {
+		return nil, status.Error(codes.PermissionDenied, "permission denied")
+	}
+
+	ctx = actor.MarkContextAsInternalActor(ctx)
+
+	pubKey, err := safe.StateGetByID[*authres.PublicKey](ctx, s.state, request.PublicKeyId)
+	if err != nil {
+		return nil, err
+	}
+
+	if pubKey.TypedSpec().Value.Identity == nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid argument")
+	}
+
+	if pubKey.TypedSpec().Value.Identity.Email != creds.Identity {
+		return nil, status.Error(codes.PermissionDenied, "permission denied")
+	}
+
+	// resetting the workload proxy cookies
+	if err = cookies.Set(
+		ctx,
+		&http.Cookie{
+			Name:     workloadproxy.PublicKeyIDCookie,
+			Path:     "/",
+			Expires:  time.Unix(0, 0),
+			HttpOnly: true,
+			Secure:   true,
+			Domain:   s.workloadProxyCookieDomain,
```

Comment:
The domain has to match the domain you want to set the cookie for, I am getting this. Is it a problem in my setup?

<img width="762" height="166" alt="Image" src="https://github.com/user-attachments/assets/d97aadf1-9ea6-4801-ac73-6d94165bb835" />

Reply:
yea, for that the domain should be `omni.local` - I imagine there is maybe some subdomain logic going on in backend but it cant set a cookie on `*.local` AFAIK. Need at least 2 parts.